runner/src/server/lib/cacheService.js

```diff
@@ -26,14 +26,14 @@ class CacheService {
 
   async clearState (request) {
     if (request.yar && request.yar.id) {
-      this.cache.drop(this.Key(request.yar.id))
+      this.cache.drop(this.Key(request.yar.id, request.query.visit))
     }
   }
 
-  Key (id) {
+  Key (sessionId, visitId) {
     return {
       segment: partition,
-      id
+      id: `${sessionId}:${visitId}`
```

Comment:
I imagine if there's no visitId the id will be `abc-ef-g:undefined`?

Reply:
Sounds about right. If that's the case then we've ballsed things up somewhere though!

Reply:
Is there something you'd prefer it was in this case?